CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/LayersViewController.java

```diff
@@ -86,13 +96,116 @@ public void submit() {
         }
 
         final List<String> layerQueries = new ArrayList<>();
-        for (Node node : pane.getLayers().getChildren()) {
+        for (final Node node : pane.getLayers().getChildren()) {
             if (GridPane.getRowIndex(node) > 0 && GridPane.getColumnIndex(node) == 2) {
                 final TextArea queryTextArea = (TextArea) node;
                 layerQueries.add(queryTextArea.getText().isBlank() ? null : queryTextArea.getText());
             }
         }
-
         PluginExecution.withPlugin(new UpdateGraphQueriesPlugin(layerQueries)).executeLater(GraphManager.getDefault().getActiveGraph());
     }
+
+    /**
+     * Executes a plugin to grab current Layers View selections and save them to
+     * the graph's Layers View State Attribute.
+     */
+    public void updateState() {
+        pane = parent.getContent();
+        PluginExecution.withPlugin(new LayersViewStateUpdater(pane, true)).executeLater(GraphManager.getDefault().getActiveGraph());
+    }
+
+    /**
+     * Executes a plugin to grab current Layers View state and load them into
+     * the Layers View UI.
+     */
+    public void loadOrCreateState() {
+        pane = parent.getContent();
+        PluginExecution.withPlugin(new LayersViewStateUpdater(pane, false)).executeLater(GraphManager.getDefault().getActiveGraph());
+    }
+
+    /**
+     * Update the display by reading and writing to/from the state attribute.
+     */
+    private static final class LayersViewStateUpdater extends SimpleEditPlugin {
+
+        private LayersViewPane pane = null;
+        private boolean isUpdateCall = false;
+
+        public LayersViewStateUpdater(final LayersViewPane pane, final boolean isUpdateCall) {
+            this.pane = pane;
+            this.isUpdateCall = isUpdateCall;
+        }
+
+        @Override
+        public void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
+            if (pane == null) {
+                return;
+            }
+            int stateAttributeId = LayersViewConcept.MetaAttribute.LAYERS_VIEW_STATE.ensure(graph);
+            // no state or on a graph update
+            if (graph.getObjectValue(stateAttributeId, 0) == null || isUpdateCall) {
```

Comment:
Since the state object is used in both the if and else blocks, you should get a reference to it above.